src/text-buffer.coffee

```diff
@@ -120,33 +120,37 @@ class TextBuffer
     @setPath(params.filePath) if params?.filePath
     @load() if params?.load
 
-  # Returns a {String} representing a unique identifier for this {TextBuffer}.
-  getId: ->
-    @id
-
-  # Called by {Serializable} mixin during deserialization.
-  deserializeParams: (params) ->
+  @deserialize: (params) ->
+    buffer = new TextBuffer(null, true)
     markerLayers = {}
     for layerId, layerState of params.markerLayers
-      markerLayers[layerId] = MarkerLayer.deserialize(this, layerState)
+      markerLayers[layerId] = MarkerLayer.deserialize(buffer, layerState)
     params.markerLayers = markerLayers
     params.defaultMarkerLayer = params.markerLayers[params.defaultMarkerLayerId]
-    params.history = History.deserialize(this, params.history)
+    params.history = History.deserialize(buffer, params.history)
     params.load = true if params.filePath
-    params
+    buffer.initialize(params)
+    buffer
```

Comment:
Removing the serializable mixin highlighted an interesting design problem, where some of our dependencies (e.g. `MarkerLayer` and `History` are cyclic). I think we might do something about it, but I didn't want to address it just yet and I simply worked it around by extracting an `initialize` method and calling it manually when deserializing. Thoughts? :thought_balloon: